cartesi-rollups/development/migration.md

```diff
@@ -1,44 +1,148 @@
 ---
 id: migration
-title: "Migrating from Sunodo CLI" 
+title: Migration guide
 ---
 
+## Migrating to Cartesi Rollups Node v1.5.x
+
+Release `v1.5.0` introduces a critical change in how epochs are closed in the Cartesi Rollups Node, transitioning from a **timestamp-based** system to a **block number-based** system.
+
+Epoch closure is now determined by the `CARTESI_EPOCH_LENGTH` environment variable (in blocks) instead of `CARTESI_EPOCH_DURATION` (in seconds).
+
+```plaintext
+CARTESI_EPOCH_LENGTH = CARTESI_EPOCH_DURATION / BLOCK_TIME
+```
+
+Where `BLOCK_TIME` is the time to generate a block in the target network.
+
+**Example**: If a block is generated every 12 seconds and `CARTESI_EPOCH_DURATION` is set to 86400 seconds (24 hours), `CARTESI_EPOCH_LENGTH = 86400 / 12 = 7200`
+
+### Option 1: Redeploy all contracts
+
+Redeploy all contracts and your application with the new configuration.
+
+For detailed deployment instructions, refer to the [self-hosted deployment guide](../deployment/self-hosted.md).
+
+:::caution
+Redeploying creates a new application instance. All previous inputs, outputs, claims, and funds locked in the application contract will remain associated with the old application address.
+:::
+
+### Option 2: Replace Application's History
```

Comment:
I missed the warning below:
https://github.com/cartesi/rollups-node/blob/08aef39cd5845aaa5abd4c21d4746f859221ea39/docs/upgrade_to_1_5.md?plain=1#L54-L56.

It's an advanced-user scenario, as such thing would be kept under the cover by sunodo. But it's a valid warning.